fuzzing/test_case_assertion_provider.go

```diff
@@ -234,3 +235,79 @@ func (t *AssertionTestCaseProvider) callSequencePostCallTest(worker *FuzzerWorke
 
 	return shrinkRequests, nil
 }
+
+// encounteredAssertionFailure takes in a panic code and a config.AssertionModesConfig and will determine whether the
+// panic code that was hit should be treated as a failing case - which will be determined by whether that panic
+// code was enabled in the config. Note that the panic codes are defined in the abiutils package and that this function
+// panic if it is provided a panic code that is not defined in the abiutils package.
+// TODO: Also note that this is a terrible design and if there is a better way to maintain assertion and panic logic, it
+//
+//	should be implemented in a future PR
+func encounteredAssertionFailure(panicCode uint64, conf config.AssertionModesConfig) bool {
+	// Switch on panic code
+	switch panicCode {
+	case abiutils.PanicCodeCompilerInserted:
+		if conf.FailOnCompilerInsertedPanic {
+			return true
+		} else {
+			return false
+		}
+	case abiutils.PanicCodeAssertFailed:
+		if conf.FailOnAssertion {
+			return true
+		} else {
+			return false
+		}
+	case abiutils.PanicCodeArithmeticUnderOverflow:
+		if conf.FailOnArithmeticUnderflow {
+			return true
+		} else {
+			return false
+		}
+	case abiutils.PanicCodeDivideByZero:
+		if conf.FailOnDivideByZero {
+			return true
+		} else {
+			return false
+		}
+	case abiutils.PanicCodeEnumTypeConversionOutOfBounds:
+		if conf.FailOnEnumTypeConversionOutOfBounds {
+			return true
+		} else {
+			return false
+		}
+	case abiutils.PanicCodeIncorrectStorageAccess:
+		if conf.FailOnIncorrectStorageAccess {
+			return true
+		} else {
+			return false
+		}
+	case abiutils.PanicCodePopEmptyArray:
+		if conf.FailOnPopEmptyArray {
+			return true
+		} else {
+			return false
+		}
+	case abiutils.PanicCodeOutOfBoundsArrayAccess:
+		if conf.FailOnOutOfBoundsArrayAccess {
+			return true
+		} else {
+			return false
+		}
+	case abiutils.PanicCodeAllocateTooMuchMemory:
+		if conf.FailOnAllocateTooMuchMemory {
+			return true
+		} else {
+			return false
+		}
+	case abiutils.PanicCodeCallUninitializedVariable:
+		if conf.FailOnCallUninitializedVariable {
+			return true
+		} else {
+			return false
+		}
+	}
+
+	// Throw a panic if we encounter a panic code not defined in the switch statement above
+	panic("encountered unknown panic code")
```

Comment:
Oh last thing, we probably don't need to panic here. If solidity adds a new panic code, we should be mindful of it to keep an eye out (hopefully the "unknown error code" message will help), but I'd say we ignore it and keep fuzzing.

I'd only panic if there was some bad code on our end where we defined some panic code type but never created a string for it, but that isn't a problem to worry about in this implementation really.

Reply:
Resolved in [cd27dcb](https://github.com/crytic/medusa/pull/161/commits/cd27dcb384e4f0ae2dc52f67534a6b329b1c1ec1)